framework/helpers/BaseHtml.php

```diff
@@ -1935,7 +1935,8 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
                         } elseif ($strict) {
                             $selected = !strcmp((string)$key, (string)$selection);
                         } else {
-                            $selected = is_object($selection) ? $selection->value == $key : $selection == $key;
+                            $selected = is_object($selection) && property_exists($selection, 'value') ?
```

Comment:
instanceof would be better

Reply:
@bizley  Can you check the latest changes